admin/pre-commit

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+
+PROJECT=`php -r "echo dirname(dirname(dirname(realpath('$0'))));"`
+STAGED_FILES_CMD=`git diff --cached --name-only --diff-filter=ACMR HEAD | grep \\\\.php`
+
+# Determine if a file list is passed
+if [ "$#" -eq 1 ]
+then
+	oIFS=$IFS
+	IFS='
+	'
+	SFILES="$1"
+	IFS=$oIFS
+fi
+SFILES=${SFILES:-$STAGED_FILES_CMD}
+
+echo "Checking PHP Lint..."
+for FILE in $SFILES
+do
+	php -l -d display_errors=0 $PROJECT/$FILE
+	if [ $? != 0 ]
+	then
+		echo "Fix the error before commit."
+		exit 1
+	fi
+	FILES="$FILES $PROJECT/$FILE"
+done
+
+if [ "$FILES" != "" ]
+then
+	echo "Running Code Sniffer..."
+	./vendor/bin/phpcbf --standard=./vendor/codeigniter4/codeigniter4-standard/CodeIgniter4 --encoding=utf-8 -n -p $FILES
+	if [ $? != 0 ]
+	then
+		echo "Fix the error before commit."
```

Comment:
CS told me that 160 errors were fixed in 1 file (that I purposely changed a few lines in), and that 82 remained.
No indication of remaining changes expected, but the file was updated.
I re-added it, and re-ran my commit, and it took it, reporting that no fixable errors were found.
Is that expected? If so, the error message might be misleading.

Reply:
Oh, no it should not exit like that. The intent of this is just to have it fix the things it can fix, which is mostly code-formatting stuff. The original article had it just checking and exiting if there were any errors. Figured that as overkill and going to put a huge damper on contributions so left changed it to just fix what it could.